sdk/src/main/java/com/bugsnag/android/DefaultDelivery.java

```diff
@@ -79,6 +80,9 @@ int deliver(String urlString,
             return conn.getResponseCode();
         } catch (IOException exception) {
             throw new DeliveryFailureException("IOException encountered in request", exception);
+        } catch (Exception exception) {
+            Logger.warn("Unexpected error delivering payload", exception);
+            return HTTP_BAD_REQUEST;
```

Comment:
Could you confirm that the `return` statement here doesn't make the `finally` block unreachable?

Reply:
Yep, as long as the catch block doesn't exit the process, the finally block will execute last.